packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1890,12 +1906,14 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
   }
 
   @override
-  double computeMinIntrinsicHeight(double width) {
-    return _preferredHeight(width);
-  }
+  double computeMinIntrinsicHeight(double width) => computeMaxIntrinsicHeight(width);
```

Comment:
hmmm not sure if this really applies in this context, computeMaxIntrinsicHeight doc says to call getMaxIntrinsicHeight instead.

Reply:
This mirrors the `RenderParagraph` implementation. In this specific case `compute{Min/Max}InsintricHeight` have the same implementation so I'm calling `computeMaxIntrinsicHeight` here. It does seem weird since I'd imagine maxIntrinsicHeight should >= minIntrinsicHeight when the width is given, but it's what RenderParagraph currently does.

Reply:
calling the same method is expected because height will be fixed once you given a width.

Reply:
getMaxIntrinsicHeight:

> Returns the smallest height beyond which increasing the height never decreases the preferred width. The preferred width is the value that would be returned by [getMinIntrinsicWidth](https://master-api.flutter.dev/flutter/rendering/RenderBox/getMinIntrinsicWidth.html) for that height.

getMinIntrinsicHeight:

> Returns the minimum height that this box could be without failing to correctly paint its contents within itself, without clipping.

I think these 2 are supposed to be using different line breaking algorithms?

By definition I think the former should be the height of the paragraph when all soft break opportunities are turned into actual line breaks (so the input widget doesn't matter):
```
hello
world
hello
world
```

and the latter should be the height of the paragraph when the max width is set to the input width, using the current greedy line breaking algorithm.
```
hello world hello world
```

Reply:
Your suggestion is changing the line break strategy (which skia txt only expose one). 
In text widget, you wouldn't expect the breaking strategy to change if you give the same width but different height constraint though

Reply:
IIRC, currently `minIntrinsicWidth` already does that (it pretends that we take every soft break opportunity)

Reply:
minIntrinsicWidth is the same line break strategy in a shorter width.

Reply:
Maybe we are talking about the same line-breaking strategy? What I meant was with `minIntrinsicWidth` the paragraph will be broken into as many lines as possible (https://github.com/google/skia/blob/85971b25cf4b94a4843f6a4e9faa6412b3d56d92/modules/skparagraph/src/TextWrapper.cpp#L434-L461).

Reply:
I guess then the question is what would user expect when the use intrinsicheight with Text.